Tools/glib/api_test_runner.py

```diff
@@ -314,11 +326,24 @@ def run_tests(self):
         # Remove skipped tests now instead of when we find them, because
         # some tests might be skipped while setting up the test environment.
         self._tests = [test for test in self._tests if self._should_run_test_program(test)]
+        number_of_qt_tests = 0
+        number_of_wpe_platform_wayland_tests = 0
+        for test in self._tests:
+            if self.is_qt_test(test):
+                number_of_qt_tests += 1
+            elif self.is_wpe_platform_wayland_test(test):
+                number_of_wpe_platform_wayland_tests += 1
+
         # Skip Qt tests if there is no GPU <https://webkit.org/b/264458>
-        number_of_qt_tests = len([test for test in self._tests if self.is_qt_test(test)])
         if number_of_qt_tests > 0 and not self._has_gpu_available():
             sys.stderr.write("WARNING: Skipping %d Qt tests because this system doesn't have a working GPU (/dev/dri devices are not available).\n" % number_of_qt_tests)
             self._tests = [test for test in self._tests if not self.is_qt_test(test)]
+
+        # Skip WPE Platform Wayland tests if weston is not available.
+        if number_of_wpe_platform_wayland_tests > 0:
+            if not WestonDriver.check_driver(self._port):
+                self._tests = [test for test in self._tests if not self.is_wpe_platform_wayland_test(test)]
```

Comment:
We should print a warning here like do when the Qt tests are skipped because of not having a working GPU